packages/composites/token-allocation/src/adapter.ts

```diff
@@ -68,35 +68,39 @@ const toValidAllocations = (allocations: any[]): TokenAllocations => {
   })
 }
 
-const computePrice = async (config: Config, allocations: TokenAllocations, quote: string) => {
+const computePrice = async (request: any, allocations: TokenAllocations, quote: string) => {
   const symbols = (allocations as TokenAllocations).map((t) => t.symbol)
-  const payload = await config.priceAdapter.getPrices(symbols, quote)
+  const payload = await request.getPrices(symbols, quote)
 
   const result = priceTotalValue(allocations, quote, payload)
   return { payload, result }
 }
 
-const computeMarketCap = async (config: Config, allocations: TokenAllocations, quote: string) => {
+const computeMarketCap = async (request: any, allocations: TokenAllocations, quote: string) => {
   const symbols = (allocations as TokenAllocations).map((t) => t.symbol)
-  const payload = await config.priceAdapter.getPrices(symbols, quote, true)
+  const payload = await request.getPrices(symbols, quote, true)
 
   const result = marketCapTotalValue(allocations, quote, payload)
   return { payload, result }
 }
 
 const inputParams = {
+  source: true,
   allocations: true,
   quote: false,
   method: false,
 }
 
 export const execute = async (input: AdapterRequest, config: Config): Promise<AdapterResponse> => {
-  const validator = new Validator(input, inputParams)
+  const paramOptions = makeOptions(config)
+  const validator = new Validator(input, inputParams, paramOptions)
   if (validator.error) throw validator.error
 
   const jobRunID = validator.validated.id
   const { quote = config.defaultQuote, method = config.defaultMethod } = validator.validated.data
   const allocations = toValidAllocations(validator.validated.data.allocations)
+  const source = validator.validated.data.source
+  const request = config.sources[source]
```

Comment:
Could we name this something like `sourceConfig`? After the type change, this should be of the `PriceAdapter` type, so we can avoid that `any` types